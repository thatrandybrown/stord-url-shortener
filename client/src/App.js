import { BrowserRouter as Router, Switch, Route } from "react-router-dom";
import { ErrorBoundary } from "./components/ErrorBoundary";
import NotFound from "./components/NotFound";
import Home from "./components/Home";
import ShortLink from "./components/ShortLink";

import "./App.css";

const App = () => {
  return (
    <>
      <header>
        <h1>STORD.AS</h1>
      </header>
      <main>
        <ErrorBoundary>
          <Router>
            <Switch>
              <Route exact path="/" children={<Home />} />
              <Route exact path="/:id" children={<ShortLink />} />
              <Route path="*" children={<NotFound />} />
            </Switch>
          </Router>
        </ErrorBoundary>
      </main>
      <footer>
        <p>&copy; 2020</p>
        <p>
          Shipping container by Creaticca Creative Agency from the Noun Project
        </p>
      </footer>
    </>
  );
};

export default App;
