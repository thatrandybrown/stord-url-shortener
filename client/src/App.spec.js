import { act, render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import App from "./App";

test("renders the app home page", () => {
  const { baseElement } = render(<App />);
  expect(baseElement).toMatchSnapshot();
});

test("landing on the id param route", async () => {
  window.history.pushState({}, "Test page", "/abcdef");

  await act(async () => {
    const { baseElement } = render(<App />, { wrapper: BrowserRouter });
    expect(baseElement).toMatchSnapshot();
  });
});

test("landing on a bad page", () => {
  window.history.pushState({}, "Test page", "/abcdef/gh");

  const { baseElement } = render(<App />, { wrapper: BrowserRouter });
  expect(baseElement).toMatchSnapshot();
});
