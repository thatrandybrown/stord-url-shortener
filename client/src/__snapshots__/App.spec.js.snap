// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`landing on a bad page 1`] = `
<body>
  <div
    id="interstitial-root"
  />
  <div>
    <header>
      <h1>
        STORD.AS
      </h1>
    </header>
    <main>
      <section
        class="error-section"
      >
        <h2
          style="font-size: 10rem;"
        >
          404
        </h2>
        <p>
          Page Not Found!
        </p>
        <a
          href="/"
        >
          Navigate Home
        </a>
      </section>
    </main>
    <footer>
      <p>
        © 2020
      </p>
      <p>
        Shipping container by Creaticca Creative Agency from the Noun Project
      </p>
    </footer>
  </div>
</body>
`;

exports[`landing on the id param route 1`] = `
<body>
  <div
    id="interstitial-root"
  >
    <div>
      <svg
        height="64"
        version="1"
        viewBox="0 0 128 128"
        width="64"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g>
          <path
            d="M78.75 16.18V1.56a64.1 64.1 0 0147.7 47.7H111.8a49.98 49.98 0 00-33.07-33.08zM16.43 49.25H1.8a64.1 64.1 0 0147.7-47.7V16.2a49.98 49.98 0 00-33.07 33.07zm33.07 62.32v14.62A64.1 64.1 0 011.8 78.5h14.63a49.98 49.98 0 0033.07 33.07zm62.32-33.07h14.62a64.1 64.1 0 01-47.7 47.7v-14.63a49.98 49.98 0 0033.08-33.07z"
            fill="#b011b0"
          />
          <animatetransform
            attributeName="transform"
            dur="400ms"
            from="0 64 64"
            repeatCount="indefinite"
            to="-90 64 64"
            type="rotate"
          />
        </g>
      </svg>
    </div>
  </div>
  <div>
    <header>
      <h1>
        STORD.AS
      </h1>
    </header>
    <main />
    <footer>
      <p>
        © 2020
      </p>
      <p>
        Shipping container by Creaticca Creative Agency from the Noun Project
      </p>
    </footer>
  </div>
</body>
`;

exports[`renders the app home page 1`] = `
<body>
  <div
    id="interstitial-root"
  />
  <div>
    <header>
      <h1>
        STORD.AS
      </h1>
    </header>
    <main>
      <p>
        Shorten your link: provide a full url and get a shortened link for sharing.
      </p>
      <form>
        <label
          for="url"
        >
          URL:
        </label>
        <input
          id="url"
          name="url"
          required=""
          type="url"
          value=""
        />
        <input
          aria-label="Shorten"
          disabled=""
          name="submit"
          type="submit"
          value="Shorten"
        />
      </form>
    </main>
    <footer>
      <p>
        © 2020
      </p>
      <p>
        Shipping container by Creaticca Creative Agency from the Noun Project
      </p>
    </footer>
  </div>
</body>
`;
